specification/automanage/resource-manager/Microsoft.Automanage/preview/2021-04-30-preview/automanage.json

```diff
@@ -1370,6 +1370,602 @@
+            "required": true,
+            "type": "string",
+            "description": "The report name."
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The report was returned successfully.",
+            "schema": {
+              "$ref": "#/definitions/Report"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "produces": [
+          "application/json"
+        ],
+        "consumes": [
+          "application/json"
+        ]
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AzureStackHci/clusters/{clusterName}/providers/Microsoft.Automanage/configurationProfileAssignments/{configurationProfileAssignmentName}/reports": {
+      "get": {
+        "tags": [
+          "HCIReports"
+        ],
+        "operationId": "HCIReports_ListByConfigurationProfileAssignments",
+        "description": "Retrieve a list of reports within a given configuration profile assignment",
+        "x-ms-examples": {
+          "List reports by HCI configuration profiles assignment": {
+            "$ref": "./examples/listReportsByconfigurationProfileHCIAssignment.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ClusterNameParameter"
+          },
+          {
+            "name": "configurationProfileAssignmentName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The configuration profile assignment name."
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. List of reports.",
+            "schema": {
+              "$ref": "#/definitions/ReportList"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
```

Comment:
>null [](http://example.com/codeflow?start=26&length=4)

same here ?

Reply:
Hi, we will implement paging and we have backlog story for that. We followed instruction as described in the documentation here:
https://armwiki.azurewebsites.net/api_contracts/guidelines/openapi.html?q=x-ms-pageable

If x-ms-pageable is added after the fact it becomes a breaking change. x-ms-pageable can be added to APIs that do not yet support pagination by setting nextLinkName to null. In this way you open yourself up for making the API paginated in the future without breaking existing clients.

Reply:
Sounds good. Thanks